youtube_dl/utils.py

```diff
@@ -183,7 +183,7 @@ def register_socks_protocols():
 ])
 
 PACKED_CODES_RE = r"}\('(.+)',(\d+),(\d+),'([^']+)'\.split\('\|'\)"
-JSON_LD_RE = r'(?is)<script[^>]+type=(["\'])application/ld\+json\1[^>]*>(?P<json_ld>.+?)</script>'
+JSON_LD_RE = r'(?is)<script[^>]+type=(["\']|)application/ld\+json\1[^>]*>(?P<json_ld>.+?)</script>'
```

Comment:
`r'(?is)<script[^>]+type=(["\']?)application/ld\+json\1[^>]*>(?P<json_ld>.+?)</script>'`.

Reply:
that will fail misserably
your condition (AKA original one) does not match when there is no quotes at all (eg. this site)
"|" will ensure match even for empty string (meaning no quotes around)
if you know about a better solution that works let me know, but original regex does not match this:
```
<script type=application/ld+json>
            {
  "@context": "http://schema.org",
  "@type": "VideoObject",
  "name": "Umělé oplodnění. Kolik stojí naděje mít vlastní dítě?",
  "description": "Naše pořady můžete nově sledovat také v aplikacích pro \r\niOS\r\n a \r\nAndroid\r\n.\r\nZplození dítěte je posvátným okamžikem, k němuž však čím dál častěji nedochází přirozenou cestou. Změny životního prostře",
  "thumbnailUrl": "https://peshkaqeni.gjirafa.com/api/media/malltv/t0g01y/retina.jpg",
  "duration": "PT15M27S",
  "uploadDate": "2019-02-01T08:30:00.0000000",
  "interactionCount": 5080
}
        </script>
```
which is what comes from the server

Reply:
No, it won't. Bother to read it carefully. `(["\']?)` is idiomatic and correct way to say `(["\']|)`.

Reply:
have it your way, it probably is more readable.